actions/java-gradle-release-github/action.yaml

```diff
@@ -0,0 +1,58 @@
+name: "Release Java artifacts"
+description: "Release Java Gradle artifacts on Github"
+
+inputs:
+  github-username:
+    description: "GitHub username for requesting changes from API."
+    required: true
+  github-token:
+    description: "GitHub token for requesting changes from API."
+    required: true
+  java-distribution:
+    description: "Java distribution to be installed. (Default is microsoft)"
+    required: false
+    default: "microsoft"
+  java-version:
+    description: "Java version to be installed. (Default is 11)"
+    required: false
+    default: "11"
+  gradle-version:
+    description: "Gradle version to be installed. (Default is wrapper)"
+    required: false
+    default: "wrapper"
+  working-directory:
+    description: "Working directory of your Gradle artifacts. (Default is .)"
+    required: false
+    default: "."
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository
+      uses: actions/checkout@v2
+
+    - name: Download build artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: build-artifact
+        path: build
+
+    - name: Set up Gradle with version ${{ inputs.gradle-version }}
+      uses: bakdata/ci-templates/actions/java-gradle-setup@feature/java-gradle
+      with:
+        java-distribution: ${{ inputs.java-distribution }}
+        java-version: ${{ inputs.java-version }}
+        gradle-version: ${{ inputs.gradle-version }}
+
+    - name: Generate changelog
+      run: ./gradlew -Pchangelog.releaseVersion=${GITHUB_REF##*/} -Pchangelog.sinceTag=${GITHUB_REF##*/} --stacktrace --info generateChangelog
```

Comment:
This action could be a general `github-release` action if we use a different method for changelog generation that does not depend on Java or Gradle

Reply:
@philipp94831 Do you think we could use something like [this action](https://github.com/heinrichreimer/action-github-changelog-generator) here instead?

Reply:
As discussed this in person, we will resolve this in a separate issue: https://github.com/bakdata/ci-templates/issues/25